Multiplayer

manuq opened the pull request:
Notes:
- This is only to handle multiple Player components and 2 players controls, like in Moddable Pong. The camera changes will be addressed separately in #9 
- The game logic didn't change. So it's a "coop": all players share the same lives and score for coins, and when one dies all of them respawn. In the future we could add other modes.

https://github.com/endlessm/moddable-platformer/issues/8

dylanmccall commented on hud.tscn:
```diff
@@ -14,15 +14,19 @@ anchor_left = 0.5
 theme_override_font_sizes/font_size = 64
-text = "Press any key
+text = "Controls
+Player left: WASD
+Player right: Arrow keys
```
I like the "player left" / "player right" scheme here, but it's awkward in the default case where we have a single player node. I'd be interested if anyone has an alternative scheme that works better, but with a similar bit of clarity over "player 1" / "player 2"?

manuq replied:
Good point, Player 1 / Player 2 is better.

dylanmccall commented on scripts/player.gd:
```diff
@@ -1,6 +1,24 @@
+	},
+	Global.Player.RIGHT:
+	{
+		"jump": "ui_up",
+		"left": "ui_left",
+		"right": "ui_right",
```
If we're defining player 2 actions in the input map, we should define player 1 actions in the input map as well instead of using the built in ui actions.

manuq replied:
Yes, let's do that. I was only copying Pong for consistency, but I can update both. The good thing about builtin actions like "ui_up" is that they are already mapped for joysticks. I will see if I can do the same for player 1 / player 2.

dylanmccall reviewed (changes requested):
Yay! Kudos for adding to the docs as well :) I have one nitpick about input map stuff, and I don't love how the Controls text is, but I realize there isn't much we can do here without a bunch of extra work.

manuq commented:
Thanks @dylanmccall . Ready for review again!
- Players are now renamed to Player 1 / Player 2 (not "left / right" like in Moddable Pong)
- Player controls are now all custom (not reusing the UI builtin actions) and should support joysticks too (untested)
- Also I found that the falling platforms were hardcoding the player name, so it didn't work for multiple players. Fixed.

I realize that it may be confusing to have a Player property in a Player scene instance. The scene came first, but it really should be renamed to "[PlayerCharacter](https://en.wikipedia.org/wiki/Player_character)", "Avatar" or something.

If this gets merged, I will do the same thing in Moddable Pong.

starnight reviewed (approved):
Thanks!  LGTM!
